encoders/nlp/TransformerTorchEncoder/__init__.py

```diff
@@ -120,9 +139,25 @@ def amp_accelerate(self):
         else:
             return nullcontext()
 
-    def _compute_embedding_from_model_output(self, outputs: 'BaseModelOutput', input_tokens: Dict):
+    def _api_call(self, query):
+        retries = 0
+
+        while retries < self.max_retries:
+            retries += 1
+            headers = {"Authorization": "Bearer " + self.api_token}
+            api_url = f"https://api-inference.huggingface.co/pipeline/feature-extraction/{self.pretrained_model_name_or_path}"
+            r = requests.post(api_url, json=query, headers=headers)
+            if r.status_code == HTTP_SERVICE_UNAVAILABLE:
+                self.logger.info('Service is currently unavailable. Model is loading.')
+                time.sleep(3)
```

Comment:
this is very long right? If the cause for this is having the model loading? should we send a DUMMY call at `post_init` to have it warm?

So later, during proper encoding, we do not wait so much

Reply:
Yes :+1: